cmd/minikube/cmd/start.go

```diff
@@ -1779,3 +1781,61 @@ func exitGuestProvision(err error) {
 	}
 	exit.Error(reason.GuestProvision, "error provisioning guest", err)
 }
+
+func startNerdctld() {
+	// create a temporary directory
+	tmpDirName, err := os.MkdirTemp("/tmp", "minikube-nerdctld")
+	if err != nil && !os.IsExist(err) {
+		exit.Error(reason.IfSSHClient, "Error with creating "+tmpDirName, err)
+	}
+
+	// for containerd runtime using ssh, we need to install nerdctld and nerdctl within os
+	if err := download.Nerdctl(tmpDirName); err != nil {
+		exit.Error(reason.IfSSHClient, "Error with downloading nerdctl", err)
+	}
+	if err := download.Nerdctld(tmpDirName); err != nil {
+		exit.Error(reason.IfSSHClient, "Error with downloading nerdctld", err)
+	}
+
+	// copy these binaries to the path of the containerd node
+	co := mustload.Running(ClusterFlagValue())
+	runner := co.CP.Runner
+	src := newRemotePath(filepath.Join(tmpDirName, "nerdctl"))
+	dst := newRemotePath("/usr/local/bin/nerdctl")
+	if err := runner.Copy(copyableFile(&co, src, dst)); err != nil {
+		exit.Error(reason.InternalCommandRunner, "Fail to copy file nerctl", err)
+	}
+
+	src = newRemotePath(filepath.Join(tmpDirName, "nerdctld"))
+	dst = newRemotePath("/usr/local/bin/nerdctld")
+	if err := runner.Copy(copyableFile(&co, src, dst)); err != nil {
+		exit.Error(reason.InternalCommandRunner, "Fail to copy file nerctld", err)
+	}
+
+	// and set 777 to these files
+	if out, err := runner.RunCmd(exec.Command("sudo", "chmod", "777", "/usr/local/bin/nerdctl", "/usr/local/bin/nerdctld")); err != nil {
+		exit.Error(reason.InternalCommandRunner, fmt.Sprintf("Failed setting permission for nerdctl: %s", out.Output()), err)
+	}
+
+	// start nerdctld
+	nerdctldDaemonCommand := exec.Command("/bin/bash", "-c", "sudo env CONTAINERD_NAMESPACE=k8s.io nohup nerdctld --socket=/var/run/nerdctld.sock >/dev/null 2>&1 &")
+	if out, err := runner.RunCmd(nerdctldDaemonCommand); err != nil {
+		exit.Error(reason.InternalCommandRunner, fmt.Sprintf("Failed to start nerdctld: %s", out.Output()), err)
+	}
+
+	// set up environment variable on remote machine. docker client uses 'non-login & non-interactive shell' therefore the only way is to modify .bashrc file of user 'docker'
+	// insert this at 4th line
+	envSetupCommand := exec.Command("/bin/bash", "-c", "sed -i '4i export DOCKER_HOST=unix:///var/run/nerdctld.sock' .bashrc")
+	if out, err := runner.RunCmd(envSetupCommand); err != nil {
+		exit.Error(reason.InternalCommandRunner, fmt.Sprintf("Failed to set up DOCKER_HOST: %s", out.Output()), err)
+	}
+
+	// need to wait for nerdctld to be started
+	time.Sleep(1 * time.Second)
```

Comment:
instead of sleeping, you could use prgrep or any other command to ensure the process is UP and running

Reply:
There is already systemd notification, when using the systemd service.